### ISSUE TITLE
Enhancement: [no-unused-vars] support explicit resource management

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#11424]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-unused-vars/

### Description

From the ESLint v9.32.0 release notes:

> [no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars): Treats variables declared with using/await using as used, since their Symbol.dispose is implicitly called at the end of their scope. Also adds an option to ignore unused using declarations.

### Fail

```TypeScript
var replace = 'me';
```

### Pass

```TypeScript
using foo = disposable();
```

**Comment by nayounsang** [comment]
Some reference to understanding this issue more...
- [no-unused-vars ignoreUsingDeclarations](https://eslint.org/docs/latest/rules/no-unused-vars#ignoreusingdeclarations)
- [explicit-resource-management](https://github.com/tc39/proposal-explicit-resource-management?tab=readme-ov-file#ecmascript-explicit-resource-management)
- [implement on eslint](https://github.com/eslint/eslint/blob/main/lib/rules/no-unused-vars.js#L373C12-L373C42)

So, we need to handle VaraiableDeclaration statements declared with `using` / `await using` and create an `ignoreUsingDeclarations` option with default value false.

**Comment by injae-kim** [comment]
Awesome fix PR: https://github.com/typescript-eslint/typescript-eslint/pull/11456 👍
